jax/experimental/jax2tf/examples/tf_js/quickdraw/quickdraw.py

```diff
@@ -0,0 +1,147 @@
+
+    x = x.reshape((x.shape[0], -1)) # flatten
+    x = nn.Dense(features=128)(x)
+    x = nn.relu(x)
+
+    x = nn.Dense(features=NB_CLASSES)(x)
+    x = nn.softmax(x)
+
+    return x
+
+def predict(params, inputs):
+  """A functional interface to the trained Module."""
+  return QuickDrawModule().apply({'params': params}, inputs)
+
+def categorical_cross_entropy_loss(logits, labels):
+  onehot_labels = common_utils.onehot(labels, logits.shape[-1])
+  return jnp.mean(-jnp.sum(onehot_labels * jnp.log(logits), axis=1))
+
+def update(optimizer, inputs, labels):
+  def loss_fn(params):
+    logits = predict(params, inputs)
+    return categorical_cross_entropy_loss(logits, labels)
+  grad = jax.grad(loss_fn)(optimizer.target)
+  optimizer = optimizer.apply_gradient(grad)
+  return optimizer
+
+def accuracy(predict: Callable, params, dataset):
+  def top_k_classes(x, k):
+    bcast_idxs = jnp.broadcast_to(np.arange(x.shape[-1]), x.shape)
+    sorted_vals, sorted_idxs = lax.sort_key_val(x, bcast_idxs)
+    topk_idxs = (
+        lax.slice_in_dim(sorted_idxs, -k, sorted_idxs.shape[-1], axis=-1))
+    return topk_idxs
+  def _per_batch(inputs, labels):
+    logits = predict(params, inputs)
+    predicted_classes = top_k_classes(logits, 1)
+    predicted_classes = predicted_classes.reshape((predicted_classes.shape[0],))
+    return jnp.mean(predicted_classes == labels)
+  batched = [_per_batch(inputs, labels) for inputs, labels in dataset]
+  return jnp.mean(jnp.stack(batched))
+
+def train_one_epoch(optimizer, train_ds):
+  for inputs, labels in train_ds:
+    optimizer = jax.jit(update)(optimizer, inputs, labels)
+  return optimizer
+
+def init_model_and_data(dir_path, classes):
+  train_ds, test_ds = load_classes(dir_path, classes)
+  rng = jax.random.PRNGKey(0)
+  init_shape = jnp.ones((1, 28, 28, 1), jnp.float32)
+  initial_params = QuickDrawModule().init(rng, init_shape)["params"]
+  optimizer = flax.optim.Adam(
+      learning_rate=0.001, beta1=0.9, beta2=0.999).create(initial_params)
+  return optimizer, initial_params, train_ds, test_ds
+
+def train(dir_path, classes):
+  optimizer, params, train_ds, test_ds = init_model_and_data(dir_path, classes)
+  for epoch in range(5):
+    start_time = time.time()
+    optimizer = train_one_epoch(optimizer, train_ds)
+    epoch_time = time.time() - start_time
+    train_acc = accuracy(predict, optimizer.target, train_ds)
+    test_acc = accuracy(predict, optimizer.target, test_ds)
+    print("Epoch {} in {:0.2f} sec".format(epoch, epoch_time))
+    print("Training set accuracy {}".format(train_acc))
+    print("Test set accuracy {}".format(test_acc))
+
+  return optimizer.target
+
+def main(*args):
+  dir_path = os.path.join(os.path.dirname(__file__), "data")
+  classes = download_dataset(dir_path, NB_CLASSES)
+  assert len(classes) == NB_CLASSES, classes
+  print(f"Classes are: {classes}")
+  flax_params = train(dir_path, classes)
+
+  model_dir = os.path.join(os.path.dirname(__file__), "saved_models")
```

Comment:
I think that it would be better to use a temporary directory for saving the models. Saving next to the sources will never work when run inside Google. I suggest either hard-coding /tmp/jax2tf/tf_js_quickdraw or `tempfile`.